File: STORE/backend/app/models/food.py
from __future__ import annotations

from typing import Literal

from app.models.base import TimestampedDocument


FoodStatus = Literal["ACTIVE", "INACTIVE"]


class FoodDocument(TimestampedDocument):
    businessId: str
    recipeId: str
    name: str
    category: str | None = None
    price: float = 0
    cost: float = 0
    estimatedProfit: float = 0
    preparationTime: int | None = None
    assignedStores: list[str] = []
    image: str | None = None
    status: FoodStatus = "ACTIVE"

File: STORE/mobile/src/api/endpoints/food.ts
import { apiClient } from '@/api/client';

export interface FoodItem {
  id: string;
  name: string;
  description?: string | null;
  category?: string | null;
  price: number;
  cost: number;
  estimatedProfit: number;
  recipeId?: string | null;
  imageUrl?: string | null;
  assignedStores?: string[];
  status?: 'ACTIVE' | 'INACTIVE' | string;
}

export async function listFood(): Promise<FoodItem[]> {
  return apiClient.get('/api/v1/food');
}

export async function getFood(id: string): Promise<FoodItem> {
  return apiClient.get(`/api/v1/food/${id}`);
}